external/js/cothority/lib/identity.js

```diff
@@ -0,0 +1,227 @@
+        //this.point.unmarshalBinary(this.pub);
+        /*return this.point.clone();*/
+        const point = group.point();
+        point.unmarshalBinary(this.pub);
+        return point;
+    }
+}
+
+
+/*
+ * Roster represents a group of servers. It basically consists in a list of
+ * ServerIdentity with some helper functions.
+ * */
+class Roster {
+
+    /*
+     * @param a list of ServerIdentity
+     * @return a Roster from the given list of identites
+     * */
+    constructor(identities,id) {
+        this._identities = identities;
+        this._id = id;
+    }
+
+    /*
+     * Random selects a random identity and returns it
+     * @return a random identity
+     * */
+    random() {
+        const idx = Math.floor(Math.random() * (this.length()-1));
+        return this._identities[idx];
+    }
+
+    /*
+     * @return the list of identities composing this Roster
+     * */
+    get identities() {
+        return this._identities;
+    }
+
+    /*
+     * @return the length of the roster
+     * */
+    get length() {
+        return this._identities.length;
+    }
+
+    /*
+     * @return the id of the roster
+     * */
+    get id() {
+        return this._id;
+    }
+
+    /**
+     * aggregateKey returns the aggregate public key for this server.
+     * It is the sum of all public keys of the identities of this Roster.
+     *
+     * @param {kyber.Group} group The group to use to compute the aggregate key.
+     * @returns {kyber.Point} The aggregate key
+     */
+    aggregateKey(group) {
+        /*if (this.aggr)*/
+            //return this.aggr;
+
+        /*this.aggr = group.point().null();*/
+        //for(var i = 0; i < this.length; i++) {
+            //this.aggr.add(aggr,this.identities[i].point(group));
+        //}
+        /*return this.aggr;*/
+
+        const aggr = group.point().null();
+        for(var i = 0; i < this.length; i++) {
+            aggr.add(aggr,this.identities[i].point(group));
+        }
+        return aggr;
+
+    }
+
+    /**
+    * Parse cothority roster toml string into a Roster object.
+    * @example
+    * // Toml needs to adhere to the following format
+    * // where public has to be a hex-encoded string.
+    *
+    *    [[servers]]
+    *        Address = "tcp://127.0.0.1:7001"
+    *        Public = "4e3008c1a2b6e022fb60b76b834f174911653e9c9b4156cc8845bfb334075655"
+    *        Description = "conode1"
+    *    [[servers]]
+    *        Address = "tcp://127.0.0.1:7003"
+    *        Public = "e5e23e58539a09d3211d8fa0fb3475d48655e0c06d83e93c8e6e7d16aa87c106"
+    *        Description = "conode2"
+    *
+    * @param {string} toml of the above format.
+    *
+    * @throws {TypeError} when toml is not a string
+    * @return {Roster} roster
+    */
+    static fromTOML(toml) {
+        if (typeof toml !== 'string')
+            throw new TypeError;
+
+        const roster = topl.parse(toml);
+        const identities = roster.servers.map((server) =>
```

Comment:
There is now a Suite line in the TOML files (if not present then suite = "Ed25519"). You can use that to look up the group by name use it to unmarshal the hex into a point.

Reply:
On the top level then ? There is not now at least.  
https://github.com/dedis/cothority/blob/master/dedis-cothority.toml
I'll try to implement the suite search functionality in kyber-js then.

Reply:
This is done at the top level currently even if it does not exists, and it defaults to edwards25519